### ISSUE TITLE
Expand things like (x+y)^2 

### ISSUE THREAD

**Issue by evykassirer** [socraticorg/mathsteps#45]
This would happen if:
- the exponent is a positive integer
- all cancelling out and most simplifications have already happened
- the base of the exponent (e.g. `x + y` in this example) *can't* be simplified further

note that issue #36 discusses the opposite of this `(x + x) * (x + x) => (x+x)^2` but contrastingly is a simplification we try early on and only happens if the "base" *can* be simplified further (e.g. x + x can be simplified to 2x)

**Comment by hmaurer** [comment]
I think there should be an extra condition that the expansion leads to a "simpler version". It seems this would be quite hard to model because the simplification might happen "10 steps ahead". For example, you might have some expression with a term like `(x+y)^5`. Expanding it might lead to a bunch of terms cancelling out and be left with something like such as `0`. However, it might also lead to a very long expression which is everything but simpler than `(x+y)^5`. This is tricky :/

**Comment by tkosan** [comment]
On the following website is a file named moses-simp.pdf that contains a paper titled "Algebraic Simplification a Guide for the Perplexed":

[https://groups.csail.mit.edu/mac/users/gjs/6.945/readings/simplification/](https://groups.csail.mit.edu/mac/users/gjs/6.945/readings/simplification/)

It describes the various ways simplification is typically used in a CAS, which may be helpful for this discussion.

**Comment by kevinbarabash** [comment]
We should probably also take into consider the context of the expression `(x+y)^2`.  If it's in an equation like `(x+y)^2 = 0` then it makes more sense to square root both sides, e.g.
```
(x+y)^2 = 0
=> nthRoot((x+y)^2) = nthRoot(0)
=> x+y = 0
=> ...
```
Having some way to detect and perform steps that apply to equations will help with this situation and others to produce steps that are more like what a human would do.

**Comment by evykassirer** [comment]
agreed @kevinbarabash - I think we're gonna have to revamp the way equations work, to be a lot more smart about things. I wonder what the steps towards that are... read some papers? have a meeting on gitter where we brainstorm stuff?

**Comment by tkosan** [comment]
@evykassirer, I think it would be useful to brainstorm about various ways mathsteps might be used in the future before determining how equations should work.

**Comment by aelnaiem** [comment]
I'll be adding a timeline and doing a better job organizing over the next few days as part of the [wiki effort](https://github.com/socraticorg/mathsteps/issues/66) and that will give us something we can discuss around

**Comment by aelnaiem** [comment]
I think this should be done within the context of the problem. So in simplifications, it should expand, and in the case of solving an equation, and the variable being solved for is in the root, then we square both sides. Seems like a decent first approach at this problem.
